### PR TITLE
[Feature] Allow the instructors to set bonus points for exercises

### PR DESCRIPTION
**Pull request by wenjie-hou**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context

Allow instructors to set a bonus point for all types of exercises. #2053

### Description

Add a bonus point field and rename the Max Scorse to Points in all types of exercises generation page.

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration
3. Click Exercise of one course
4. Click generate new programming exercise/ Create new modeling exercise/ create new file upload exercise/ create new text exercise.
5. set bonus point for the new exercise

### Screenshots

fileupload
text
modeling
programming

### CODE REVIEW

File: src/main/webapp/app/exercises/file-upload/manage/file-upload-exercise-update.component.html [hunk 1]
```diff
@@ -64,19 +64,39 @@ <h2 *ngIf="!fileUploadExercise.id" id="jhi-file-upload-exercise-heading-create"
...
+                        <div class="form-group">
+                            <label class="form-control-label" jhiTranslate="artemisApp.exercise.points" for="field_points">Points</label>
+                            <input
+                                required
+                                type="number"
+                                class="form-control"
+                                [pattern]="maxScorePattern"
+                                min="0"
+                                max="9999"
+                                name="maxScore"
```
Comment (derLalla) [inline src/main/webapp/app/exercises/file-upload/manage/file-upload-exercise-update.component.html:78]: Shouldn't be the name set to points?

```suggestion
                                name="points"
```

File: src/main/webapp/app/exercises/file-upload/manage/file-upload-exercise-update.component.html [hunk 2]
```diff
@@ -64,19 +64,39 @@ <h2 *ngIf="!fileUploadExercise.id" id="jhi-file-upload-exercise-heading-create"
...
+                        <div class="form-group">
+                            <label class="form-control-label" jhiTranslate="artemisApp.exercise.bonusPoints" for="field_bonusPoints">Bonus Points</label>
+                            <input
+                                required
+                                type="number"
+                                class="form-control"
+                                [pattern]="maxScorePattern"
+                                min="0"
+                                max="9999"
+                                name="maxScore"
```
Comment (derLalla) [inline src/main/webapp/app/exercises/file-upload/manage/file-upload-exercise-update.component.html:94]:
```suggestion
                                name="bonusPoints"
```

File: src/main/webapp/app/exercises/text/manage/text-exercise/text-exercise-update.component.html [hunk 3]
```diff
@@ -60,19 +60,39 @@ <h4 *ngIf="isImport" id="jhi-text-exercise-heading-import" jhiTranslate="artemis
...
+                <div class="form-group">
+                    <label class="form-control-label" jhiTranslate="artemisApp.exercise.points" for="field_points">Points</label>
+                    <input
+                        required
+                        type="number"
+                        class="form-control"
+                        [pattern]="maxScorePattern"
+                        min="0"
+                        max="9999"
+                        name="maxScore"
```
Comment (derLalla) [inline src/main/webapp/app/exercises/text/manage/text-exercise/text-exercise-update.component.html:74]:
```suggestion
                        name="points"
```

File: src/main/webapp/app/exercises/text/manage/text-exercise/text-exercise-update.component.html [hunk 4]
```diff
@@ -60,19 +60,39 @@ <h4 *ngIf="isImport" id="jhi-text-exercise-heading-import" jhiTranslate="artemis
...
+                    <label class="form-control-label" jhiTranslate="artemisApp.exercise.bonusPoints" for="field_bonusPoints">Bonus Points</label>
+                    <input
+                        required
+                        type="number"
+                        class="form-control"
+                        [pattern]="maxScorePattern"
+                        min="0"
+                        max="9999"
+                        name="maxScore"
+                        id="bonusPoints"
```
Comment (derLalla) [inline src/main/webapp/app/exercises/text/manage/text-exercise/text-exercise-update.component.html:91]:
```suggestion
                        name="bonusPoints"
                        id="field_bonusPoints"
```

**Review by derLalla — CHANGES_REQUESTED** [review]

File: src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.html [hunk 5]
```diff
@@ -127,25 +127,51 @@ <h4 *ngIf="isImport" id="jhi-programming-exercise-heading-import" jhiTranslate="
...
+                                min="1"
+                                max="9999"
+                                name="points"
+                                id="field_points"
+                                [(ngModel)]="programmingExercise.maxScore"
+                                #maxScore="ngModel"
+                            />
+                            <ng-container *ngFor="let e of maxScore.errors! | keyvalue | removekeys: ['required']">
+                                <div *ngIf="maxScore.invalid && (maxScore.dirty || maxScore.touched)" class="alert alert-danger">
+                                    <div [jhiTranslate]="'artemisApp.exercise.form.maxScore' + '.' + e.key"></div>
```
Comment (schultek) [inline src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.html:148]: This needs to be points now.
```suggestion
                                    <div [jhiTranslate]="'artemisApp.exercise.form.points' + '.' + e.key"></div>
```

File: src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.html [hunk 6]
```diff
@@ -127,25 +127,51 @@ <h4 *ngIf="isImport" id="jhi-programming-exercise-heading-import" jhiTranslate="
...
+                                required
+                                type="number"
+                                [pattern]="maxScorePattern"
+                                class="form-control"
+                                min="1"
+                                max="9999"
+                                name="bonusPoints"
+                                id="field_bonusPoints"
+                                [(ngModel)]="programmingExercise.bonusPoints"
+                                #maxScore="ngModel"
```
Comment (schultek) [inline src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.html:166]: Rename this to bonusPoints
```suggestion
                                #bonusPoints="ngModel"
```

File: src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.html [hunk 7]
```diff
@@ -127,25 +127,51 @@ <h4 *ngIf="isImport" id="jhi-programming-exercise-heading-import" jhiTranslate="
...
+                            />
+                            <ng-container *ngFor="let e of maxScore.errors! | keyvalue | removekeys: ['required']">
+                                <div *ngIf="maxScore.invalid && (maxScore.dirty || maxScore.touched)" class="alert alert-danger">
```
Comment (schultek) [inline src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.html:169]: Rename this to bonusPoints
```suggestion
                            <ng-container *ngFor="let e of bonusPoints.errors! | keyvalue | removekeys: ['required']">
                                <div *ngIf="bonusPoints.invalid && (bonusPoints.dirty || bonusPoints.touched)" class="alert alert-danger">
```

File: src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.html [hunk 8]
```diff
@@ -127,25 +127,51 @@ <h4 *ngIf="isImport" id="jhi-programming-exercise-heading-import" jhiTranslate="
...
+                                    <div [jhiTranslate]="'artemisApp.exercise.form.points' + '.' + e.key"></div>
```
Comment (schultek) [inline src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.html:170]: This needs to be bonusPoints.
```suggestion
                                    <div [jhiTranslate]="'artemisApp.exercise.form.bonusPoints' + '.' + e.key"></div>
```

**Review by schultek — CHANGES_REQUESTED** [review]

**Review by schultek — APPROVED** [review]

**Review by krusche — APPROVED** [review]

**Review by anditurdiu — APPROVED** [review]
Works as expected. Good job :)

**Review by FrankeLukas — APPROVED** [review]

**Pull request merged**